runtime/shared_common/shrinit.cpp

```diff
@@ -4245,6 +4316,18 @@ j9shr_shutdown(J9JavaVM *vm)
 		if (utfHashTable) {
 			hashTableFree(utfHashTable);
 		}
+		if (NULL != romToRamHashTable) {
+			omrthread_rwmutex_enter_write(romToRamHashTableMutex);
+			if (romToRamHashTable) {
+				hashTableFree(romToRamHashTable);
+			}
+			omrthread_rwmutex_exit_write(romToRamHashTableMutex);
+
+			omrthread_rwmutex_destroy(romToRamHashTableMutex);
+
+			config->romToRamHashTable = NULL;
+			config->romToRamHashTableMutex = NULL;
```

Comment:
Better to move these 2 lines right after `omrthread_rwmutex_enter_write(romToRamHashTableMutex);`